cart/domain/cart/giftcard.go

```diff
@@ -99,3 +99,14 @@ func (cards *AppliedGiftCards) ByRemaining() AppliedGiftCards {
 	}
 	return result
 }
+
+// GiftCardByCode returns a single gift card if the given code matches its code.
+func (cards *AppliedGiftCards) GiftCardByCode(code string) (card AppliedGiftCard, found bool) {
+	for _, card = range *cards {
+		if card.Code == code {
+			return card, true
+		}
+	}
+
+	return AppliedGiftCard{}, false
```

Comment:
why not nil, false (change to pointer) 
I know that we said that dealing with Values objects over References - but imho thats more relevant for parameter in called methods.
In case something is optional I find its more logic to return nil instead an unfilled struct of that type. We do it at other places as well this way

Reply:
Yeah, that makes sense. See: https://github.com/i-love-flamingo/flamingo-commerce/pull/160